app/Http/Controllers/HomeController.php

```diff
@@ -3,82 +3,47 @@
+use App\Repositories\Contracts\CategoryRepository;
+use App\Repositories\Contracts\BookRepository;
+use App\Eloquent\Bookmeta;
 
 class HomeController extends Controller
 {
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index()
-    {
-        return view('admin.layout.index');
-    }
-
-    /**
-     * Show the form for creating a new resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function create()
-    {
-        //
-    }
-
-    /**
-     * Store a newly created resource in storage.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
-     */
-    public function store(Request $request)
-    {
-        //
-    }
+    protected $bookRepository;
 
-    /**
-     * Display the specified resource.
-     *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
-    public function show($id)
-    {
-        //
-    }
+    protected $officeRepository;
 
-    /**
-     * Show the form for editing the specified resource.
-     *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
-    public function edit($id)
-    {
-        //
-    }
+    protected $imageProperties = [
+        'path',
+        'target_id',
+    ];
 
-    /**
-     * Update the specified resource in storage.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
-    public function update(Request $request, $id)
+    public function __construct(BookRepository $bookRepository, OfficeRepository $officeRepository)
     {
-        //
+        $this->bookRepository = $bookRepository;
+        $this->officeRepository = $officeRepository;
     }
 
-    /**
-     * Remove the specified resource from storage.
-     *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
-    public function destroy($id)
+    public function index()
     {
-        //
+        $with = [
+            'medias' => function ($q) {
+                $q->select($this->imageProperties);
+            },
+        ];
+        $offices = $this->officeRepository->getData()->pluck('name', 'id');
+        $topViewed = $this->bookRepository->getTopViewedBook($with);
+        $topReview = $this->bookRepository->getTopReviewBook($with);
+        $topInteresting = $this->bookRepository->getTopInterestingBook($with);
+        $latestBook = $this->bookRepository->getLatestBook($with);
+        $bestSharing = $this->bookRepository->getBestSharing($with);
+        $hotUser = $bestSharing['user'];
+        $bestSharing = $bestSharing['books'];
+        $officeBooks = $this->bookRepository->getOfficeBooks($offices, $with);
+        
+        return view(
+            'index',
+            compact('topInteresting', 'topReview', 'topViewed', 'latestBook', 'bestSharing', 'hotUser', 'officeBooks'));
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself